src/sst/core/testingframework/sst_unittest.py

```diff
@@ -0,0 +1,310 @@
+           :param: err_file (str): The FilePath to the finalized error file
+                                   The default is the same as the output file.
+           :param: mpi_out_files (str): The FilePath to the mpi run output files
+                                        These will be merged into the out_file at
+                                        the end of a multi-rank run
+           :param: other_args (str): Any other arguments used in the SST cmd
+                                     that the caller wishes to use
+           :param: num_ranks (int): The number of ranks to run SST with
+           :param: num_threads (int): The number of threads to run SST with
+           :param: global_args (str): Global Arguments provided from test engine args
+           :param: timeout_sec (int|float): Allowed runtime in seconds
+        """
+        # NOTE: We cannot set the default of param to the global variable due to
+        # oddities on how this class loads, so we do it here.
+        if num_ranks == None:
+            num_ranks=test_engine_globals.TESTENGINE_SSTRUN_NUMRANKS
+        if num_threads == None:
+            num_threads=test_engine_globals.TESTENGINE_SSTRUN_NUMTHREADS
+        if global_args == None:
+            global_args=test_engine_globals.TESTENGINE_SSTRUN_GLOBALARGS
+
+        # Make sure arguments are of valid types
+        check_param_type("sdl_file", sdl_file, str)
+        check_param_type("out_file", out_file, str)
+        if err_file is not None:
+            check_param_type("err_file", out_file, str)
+        if set_cwd is not None:
+            check_param_type("set_cwd", set_cwd, str)
+        check_param_type("mpi_out_files", mpi_out_files, str)
+        check_param_type("other_args", other_args, str)
+        if num_ranks is not None:
+            check_param_type("num_ranks", num_ranks, int)
+        if num_threads is not None:
+            check_param_type("num_threads", num_threads, int)
+        if global_args is not None:
+            check_param_type("global_args", global_args, str)
+        if not (isinstance(timeout_sec, (int, float)) and not isinstance(timeout_sec, bool)):
+            raise ValueError("ERROR: Timeout_sec must be a postive int or a float")
+
+        # Make sure sdl file is exists and is a file
+        if not os.path.exists(sdl_file) or not os.path.isfile(sdl_file):
+            log_error("sdl_file {0} does not exist".format(sdl_file))
+
+        # Figure out a name for the mpi_output files if the default is provided
+        if mpi_out_files == "":
+            mpiout_filename = "{0}.testfile".format(out_file)
+        else:
+            mpiout_filename = mpi_out_files
+
+        # Set the initial os launch command for sst.
+        # If multi-threaded, include the number of threads
+        if num_threads > 1:
+            oscmd = "sst -n {0} {1} {2} {3}".format(num_threads,
+                                                    global_args,
```

Comment:
Same as above.  This code has been solidly working for a while now.  While it could be better with an array, we would need to implement the whole OSCommand class to use arrays.  Recommend we do this for a future release.